NUthreads.Infrastructure/Repositories/UserRepository.cs

```
@@ -1,71 +1,45 @@
 ﻿using Microsoft.EntityFrameworkCore;
 using MongoDB.Driver;
+using NUthreads.Application.Interfaces.Repositories;
+using NUthreads.Domain.DTOs;
 using NUthreads.Domain.Models;
 using NUthreads.Infrastructure.Contexts;
+using NUthreads.Infrastructure.Repositories.Common;
 
 
 namespace NUthreads.Infrastructure.Repositories
 {
-    public class UserRepository : IUserRepository
+    public class UserRepository : BaseRepository<User>, IUserRepository
     {
-        private readonly IMongoCollection<User> _users;
+        private readonly NUthreadsDbContext _context;
+        private readonly DbSet<User> _users;
 
-        public UserRepository(IMongoClient mongoClient)
+        public UserRepository(NUthreadsDbContext context) : base(context)
         {
-            var database = mongoClient.GetDatabase("NUthreadsDB");
-            _users = database.GetCollection<User>("Users");
+            _context = context;
```

null check here please.
whenever you inject always ask yourself so that you never forget, would i want to throw an exception in the runtime or catch it early when building?